src/Compat.jl

```diff
@@ -1891,6 +1891,17 @@ else
     range(start::Number, stop::Number; kwargs...) = range(start; stop=stop, kwargs...)
 end
 
+# https://github.com/JuliaLang/julia/pull/30496
+if VERSION < v"1.2.0-DEV.272"
+    Base.@pure hasfield(::Type{T}, name::Symbol) where T =
+        Base.fieldindex(T, name, false) > 0
+    export hasfield
+    if VERSION >= v"0.7-"
```

Comment:
```suggestion
    if VERSION >= v"0.7-DEV.5136"
```
That's the version that exports `propertynames`. But doesn't really matter. Whether Compat defines `hasproperty` in a way that errors or not at all on early `0.7-DEV` is unlikely to ever make a difference.

I do wonder, however, whether we should add `hasproperty(x, s::Symbol) = hasfield(typeof(x), s)` for old versions. (Or maybe `propertynames(x) = fieldnames(typeof(x))`.) Then one can use `hasproperty` on any version to check whether `x.s` is expected to work.

Reply:
Now that both 0.7 and 0.6 are unmaintained, is it worth complicating Compat.jl?  Also, isn't it better to wait until someone requests it (or rather implements it and adds tests for it)?

Reply:
True. I think it's still a bit early to drop the 0.6-stuff, but not worth worrying too much when adding new things.